xgi/utils/utilities.py

```diff
@@ -96,18 +96,22 @@ def update_uid_counter(H):
     ----------
     H : xgi.Hypergraph
         Hypergraph of which to update the uid counter
+    id : any hashable type
+        User-provided ID.
 
     """
-
-    edges_with_int_id = [
-        int(e)
-        for e in H.edges
-        if (not isinstance(e, str))
-        and not isinstance(e, tuple)
-        and float(e).is_integer()
-    ]
-
-    # Then, we set the start at one plus the maximum edge ID that is an integer,
-    # because count() only yields integer IDs.
-    start = max(edges_with_int_id) + 1 if edges_with_int_id else 0
+    old_id = next(H._edge_uid)
```

Comment:
It took me longer than felt right to understand to be fully convinced by the code 😅 I think it's because of the naming `old_id`. Do you think something like `potential_new_id` or `default_id` or even `new_id` would work? It's only old if it's not selected hehe.

Reply:
Haha good call - I will fix this :)